components/cam/src/dynamics/se/dp_coupling.F90

```diff
@@ -1,279 +1,280 @@
-             ie = idmb3(1)
-             ioff=idmb2(1)
-
-             phys_state(lchnk)%ps(icol)=ps_tmp(ioff,ie)
-             phys_state(lchnk)%phis(icol)=phis_tmp(ioff,ie)
-             do ilyr=1,pver
-                phys_state(lchnk)%t(icol,ilyr)=T_tmp(ioff,ilyr,ie)	   
-                phys_state(lchnk)%u(icol,ilyr)=uv_tmp(ioff,1,ilyr,ie)
-                phys_state(lchnk)%v(icol,ilyr)=uv_tmp(ioff,2,ilyr,ie)
-                phys_state(lchnk)%omega(icol,ilyr)=omega_tmp(ioff,ilyr,ie)
-
-                if (use_gw_front) then
-                   pbuf_frontgf(icol,ilyr) = frontgf(ioff,ilyr,ie)
-                   pbuf_frontga(icol,ilyr) = frontga(ioff,ilyr,ie)
-                endif
-             end do
-
-             do m=1,pcnst
-                do ilyr=1,pver
-                   phys_state(lchnk)%q(icol,ilyr,m)=Q_tmp(ioff,ilyr,m,ie)
-                end do
-             end do
-          end do
-       end do
+      !$omp parallel do private (lchnk, ncols, pgcols, icol, idmb1, idmb2, idmb3, ie, ioff, ilyr, m, pbuf_chnk, pbuf_frontgf, pbuf_frontga)
+      do lchnk = begchunk,endchunk
+        ncols=get_ncols_p(lchnk)
+        call get_gcol_all_p(lchnk,pcols,pgcols)
+
+        pbuf_chnk => pbuf_get_chunk(pbuf2d, lchnk)
+
+        if (use_gw_front) then
+          call pbuf_get_field(pbuf_chnk, frontgf_idx, pbuf_frontgf)
+          call pbuf_get_field(pbuf_chnk, frontga_idx, pbuf_frontga)
+        end if
+
+        do icol = 1,ncols
+          call get_gcol_block_d(pgcols(icol),1,idmb1,idmb2,idmb3)
+          ie = idmb3(1)
+          ioff = idmb2(1)
+          phys_state(lchnk)%ps(icol)   = ps_tmp(ioff,ie)
+          phys_state(lchnk)%phis(icol) = zs_tmp(ioff,ie)
+          do ilyr = 1,pver
+            phys_state(lchnk)%t(icol,ilyr)     = T_tmp(ioff,ilyr,ie)	   
+            phys_state(lchnk)%u(icol,ilyr)     = uv_tmp(ioff,1,ilyr,ie)
+            phys_state(lchnk)%v(icol,ilyr)     = uv_tmp(ioff,2,ilyr,ie)
+            phys_state(lchnk)%omega(icol,ilyr) = w_tmp(ioff,ilyr,ie)
+            if (use_gw_front) then
+              pbuf_frontgf(icol,ilyr) = frontgf(ioff,ilyr,ie)
+              pbuf_frontga(icol,ilyr) = frontga(ioff,ilyr,ie)
+            end if
+          end do ! ilyr
+
+          do m = 1,pcnst
+            do ilyr = 1,pver
+              phys_state(lchnk)%q(icol,ilyr,m) = q_tmp(ioff,ilyr,m,ie)
+            end do ! ilyr
+          end do ! m
+        end do ! icol
+      end do ! lchnk
 
     else  ! .not. local_dp_map
 
-       tsize = 4 + pcnst
-       if (use_gw_front) tsize = tsize + 2
+      tsize = 4 + pcnst
+      if (use_gw_front) tsize = tsize + 2
 
-       allocate(bbuffer(tsize*block_buf_nrecs))
-       allocate(cbuffer(tsize*chunk_buf_nrecs))
-       if(par%dynproc) then
-!$omp parallel do private (ie, bpter, icol, ilyr, m)
-          do ie=1,nelemd
+      allocate(bbuffer(tsize*block_buf_nrecs))
+      allocate(cbuffer(tsize*chunk_buf_nrecs))
+      allocate(bpter(nphys_sq,0:pver))
```

Comment:
Another allocate (and 1 more is in p_d_coupling) which can be static?

Reply:
oh yea, I forgot about this one. I'll change it.

Reply:
In case this was forgotten: do you still plan to change this allocate and 1 more matching in p_d_coupling?

Reply:
You probably noticed this already, but I did revert this change in a commit that I pushed yesterday.

Reply:
Thanks, sorry I missed it, I see it now. I opened the whole file to track it today but it must have been from old history tree.